webpay/pay/views.py

```diff
@@ -412,7 +413,7 @@ def trans_start_url(request):
         with statsd.timer('purchase.payment_time.get_transaction'):
             trans = solitude.get_transaction(trans_id)
         data['status'] = trans['status']
-        data['provider'] = constants.PROVIDERS_INVERTED[trans['provider']]
+        data['provider'] = constants.PROVIDERS_INVERTED.get(trans['provider'])
```

Comment:
wouldn't this put some confusing data in our DB if we forget to map a new provider? A KeyError might be the right thing here.